plugin-enclave-gradle/src/main/kotlin/com/r3/conclave/plugin/enclave/gradle/gramine/GenerateGramineDirectManifest.kt

```diff
@@ -64,17 +69,19 @@ open class GenerateGramineManifest @Inject constructor(
          * Because of this, we need to add some safety margin.
          */
         val enclaveWorkerThreadCount = maxThreads.get()
-        val gramineMaxThreads = enclaveWorkerThreadCount + 8
+        //  TODO: Create a new parameters to distinguish between proper Enclave threads and threads used in the Enclave/Host Interface
+        val gramineMaxThreads = enclaveWorkerThreadCount * 2
```

Comment:
Why does it need to be doubled?

Reply:
This change allows us to run the code, otherwise it won't work.

We need a specific number of threads to run enclave code in Gramine (i.e. Python).
In addition to that we need another set of threads to handle the communication enclave/host.
With the previous code we only allowed 8 threads to be actually used inside the Enclave.
You need to consider that Gramine takes at least 3 threads for other operations, which means that we only had 5 threads to run Python stuff.

CC @ac101m

Reply:
We shouldn't be taking the user input of `maxThreads` and then modifying it in mysterious ways hidden from the user. Instead it is better to have sensible default values that should work in most cases. See https://r3-cev.atlassian.net/browse/CON-1223. There's no need to do anything right now.

Reply:
To elaborate, because we're creating threads greedily in the call interface, some of the threads are eaten up by that (bar 8), which will potentially prevent the user from creating threads of their own.

This isn't the whole story though, I've actually discovered some additional complexity here while looking into CON-360. It turns out gramine is actually able to re-use thread control blocks when a thread yields, meaning the number of threads that can conceptually exist in the enclave isn't actually limited to the TCB count, just the number of running threads.

Not sure exactly what should be done about it, but doubling ensures that the level of concurrency available should be roughly equal to the value in the build.gradle regardless of how the user decides to do it (either creating threads in the enclave or calling the enclave concurrently from the host).

Should also probably implement your previous suggestion of not greedily creating threads/connections in the call interface!